src/util/Utils.h

```diff
@@ -103,6 +103,49 @@ class Utils {
     static std::string checkFlag(std::string flagPath);
 
     static int connect_wrapper(int sock, const sockaddr *addr, socklen_t slen);
+
+    /**
+     * Wrapper to recv(2) to read a string.
+     *
+     * @param connFd connection file descriptor
+     * @param buf writable buffer of size at least len+1
+     * @param len maximum length of the string to read (excluding null terminator)
+     * @param allowEmpty whether to allow reading empty string or not
+     * @return The string read or "" on error. Logs error if recv failed. Also logs error if allowEmpty is false and
+     * read length 0 string.
+     */
+    static std::string read_str_wrapper(int connFd, char *buf, size_t len, bool allowEmpty);
+
+    /**
+     * Wrapper to recv(2) to read a string and trim it.
+     *
+     * @param connFd connection file descriptor
+     * @param buf writable buffer of size at least len+1
+     * @param len maximum length of the string to read (excluding null terminator)
+     * @return The trimmed string read or "" on error. Also rerurns "" if read length 0 string. This may return an
+     * empty string if the string read is non-empty but contained only white-space characters.
+     */
+    static std::string read_str_trim_wrapper(int connFd, char *buf, size_t len);
+
+    /**
+     * Wrapper to send(2) to send a data to socket.
+     *
+     * @param connFd connection file descriptor
+     * @param buf readable buffer of size at least `size`
+     * @param size size of data to send
+     * @return true on success or false otherwise. Logs error if send() failed or sent less than the requested size.
+     */
+    static bool send_wrapper(int connFd, const char *buf, size_t size);
+
+    /**
+     * Wrapper to send(2) to send a data to socket.
```

Comment:
This description ` Wrapper to send(2) to send a data to socket.` is very similar to the earlier function's description. Can we provide more details and be more specific so that reader can understand the difference between the two functions?

Reply:
Updated doc for the 2 methods

Reply:
Ok